### ISSUE TITLE
 Not Found for url: http://192.168.0.10:9015/api/libraries/9c5d18ac-a65a-4dcb-acbb-b4164adcfcc0/removeOffline

### ISSUE THREAD

**Issue by picarica** [Salvoxia/immich-folder-album-creator#67]
so i have little weird settup i am getting these errors

```
time=2024-10-07T12:10:00.253+02:00 level=INFO msg=Detected Immich server version 1.117.0
time=2024-10-07T12:10:00.253+02:00 level=INFO msg=Requesting all assets
time=2024-10-07T12:10:00.366+02:00 level=INFO msg=1609 photos found
time=2024-10-07T12:10:00.366+02:00 level=INFO msg=Sorting assets to corresponding albums using folder name
time=2024-10-07T12:10:00.368+02:00 level=INFO msg=0 albums identified
time=2024-10-07T12:10:00.368+02:00 level=INFO msg=Album list: []
time=2024-10-07T12:10:00.368+02:00 level=INFO msg=Listing existing albums on immich
time=2024-10-07T12:10:00.374+02:00 level=INFO msg=0 existing albums identified
time=2024-10-07T12:10:00.374+02:00 level=INFO msg=Creating albums if needed
time=2024-10-07T12:10:00.374+02:00 level=INFO msg=0 albums created
time=2024-10-07T12:10:00.374+02:00 level=INFO msg=Adding assets to albums
time=2024-10-07T12:10:00.374+02:00 level=INFO msg=Trigger offline asset removal
time=2024-10-07T12:10:00.380+02:00 level=ERROR msg=Error in API call: {'message': 'Cannot POST /api/libraries/9c5d18ac-a65a-4dcb-acbb-b4164adcfcc0/removeOffline', 'error': 'Not Found', 'statusCode': 404, 'correlationId': 'wfyt3bkb'}
Traceback (most recent call last):
  File "/script/immich_auto_album.py", line 1041, in <module>
    triggerOfflineAssetRemoval(library["id"])
  File "/script/immich_auto_album.py", line 637, in triggerOfflineAssetRemoval
    checkApiResponse(r)
  File "/script/immich_auto_album.py", line 709, in checkApiResponse
    response.raise_for_status()
  File "/usr/local/lib/python3.12/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://192.168.0.10:9015/api/libraries/9c5d18ac-a65a-4dcb-acbb-b416
4adcfcc0/removeOffline
```

so i have a setup that my photos are in Truenas where they are uploaded to nextcloud, and then  i just mount it to immich server where they are set as external library, so i made new docker compose file mounted the same photo library as the immich and tried to run it but got the erro above, this is my docker compose file 

```

services:  
  immich-folder-album-creator:
    container_name: immich_folder_album_creator
    image: salvoxia/immich-folder-album-creator:latest
    restart: unless-stopped
    volumes:
      - "/mnt/.ix-apps/app_mounts/nextcloud/data/picarica/files/Albumy\ mobil/:/external_libs/photos"
    environment:
      API_URL: http://192.168.0.10:9015/api
      API_KEY: Wf___REDACTED__4
      ROOT_PATH: /external_libs/photos
      # Run 10 minutes past the hour to give Offline Asset Removal time to work
      CRON_EXPRESSION: "10 * * * *"
      TZ: Europe/Bratislava
      # Delete empty albums after each run
      SYNC_MODE: "2"

```

i am not really sure what am i doing wrong, it doesnt identify the albums
when i enter the  docker container i see albums just fine

`docker exec -it immich_folder_album_creator sh`

```
/external_libs/photos # ls
Adventures                Important                 archive.zip                     redacted.zip      reyko                     zbytok zver.
Camera                    Progress                  anička                    fam                       sisa
CapCut                    Scenery                   tedacted.tar  my ca                     some dude
redacted              redacted.tar               bike baby                 ninka                     redacted.7z

```
and each of these folder have corresnding image, is that not how it should work ?

**Comment by Salvoxia** [comment]
Hi,

there seem to be two separate issues. Let's start with the easire one:

1. Error message `404 Client Error: Not Found for url: http://192.168.0.10:9015/api/libraries/9c5d18ac-a65a-4dcb-acbb-b416`
That is due to an API change in Immich Server version v1.116.0. The latest script version with tag `0.13.0` implements a fix for this (refer to [release notes for v0.13.0](https://github.com/Salvoxia/immich-folder-album-creator/releases/tag/0.13.0)). You seeing this error indicates that your local `latest` tag is not up to date. Please force a re-pull of the `latest` tag or switch to the name tag `0.13.0` in your compose file.

2.  No Albums getting identified
Looking at your compose configuration, there seems to be a misunderstand. You don't need to mount your image library into the script's container. The script finds images by querying the Immich API, so Immich must know your library. And your `ROOT_PATH` must be set to the same path your library is mounted to in the Immich container. Can you confirm that this as  ´/external_libs/photos` as well?
If not, please change your `ROOT_PATH` accordingly and retry.

Best Regards,
Salvoxia

**Comment by picarica** [comment]
yes it was my bad, my ext library on my server was on mounted /ext , but i just used default for album-fix when i changed the paths and removed mounting of the path from immich-album-fix it started working, thank you for clarifications
